examples/platforms/efr32/radio.c

```diff
@@ -0,0 +1,843 @@
+        false,                                   // isPanCoordinator
+        RAIL_IEEE802154_ACCEPT_STANDARD_FRAMES,  // framesMask
+        RAIL_RF_STATE_RX,                        // defaultState
+        100,                                     // idleTime
+        192,                                     // turnaroundTime
+        894,                                     // ackTimeout
+        NULL                                     // addresses
+    };
+
+    sReceiveFrame.mLength  = 0;
+    sReceiveFrame.mPsdu    = sReceivePsdu;
+    sTransmitFrame.mLength = 0;
+    sTransmitFrame.mPsdu   = sTransmitPsdu;
+
+    if (RAIL_IEEE802154_2p4GHzRadioConfig())
+    {
+        assert(false);
+    }
+
+    if (RAIL_IEEE802154_Init(&config))
+    {
+        assert(false);
+    }
+
+    RAIL_TxPowerSet(0);
+
+    otLogInfoPlat(sInstance, "Initialized", NULL);
+}
+
+void efr32RadioDeinit(void)
+{
+    RAIL_RfIdle();
+    RAIL_IEEE802154_Deinit();
+}
+
+void setChannel(uint8_t channel)
+{
+    if (sChannel != channel)
+    {
+        bool enabled = false;
+
+        if (sIsReceiverEnabled)
+        {
+            RAIL_RfIdle();
+            enabled = true;
+            sIsReceiverEnabled = false;
+        }
+
+        otLogInfoPlat(sInstance, "Channel=%d", channel);
+
+        sChannel = channel;
+
+        if (enabled)
+        {
+            if (RAIL_RxStart(channel))
+            {
+                assert(false);
+            }
+
+            sIsReceiverEnabled = true;
+        }
+    }
+}
+
+void otPlatRadioGetIeeeEui64(otInstance *aInstance, uint8_t *aIeeeEui64)
+{
+    uint64_t eui64;
+    uint8_t *eui64Ptr = NULL;
+    (void)aInstance;
+
+    eui64 = SYSTEM_GetUnique();
+    eui64Ptr = (uint8_t *)&eui64;
+
+    for (uint8_t i = 0; i < OT_EXT_ADDRESS_SIZE; i++)
+    {
+        aIeeeEui64[i] = eui64Ptr[(OT_EXT_ADDRESS_SIZE - 1) - i];
+    }
+}
+
+void otPlatRadioSetPanId(otInstance *aInstance, uint16_t aPanId)
+{
+    (void)aInstance;
+
+    otLogInfoPlat(sInstance, "PANID=%X", aPanId);
+
+    sPanId = aPanId;
+    RAIL_IEEE802154_SetPanId(aPanId);
+}
+
+void otPlatRadioSetExtendedAddress(otInstance *aInstance, uint8_t *aAddress)
+{
+    (void)aInstance;
+
+    otLogInfoPlat(sInstance, "ExtAddr=%X%X%X%X%X%X%X%X",
+                  aAddress[7], aAddress[6], aAddress[5], aAddress[4],
+                  aAddress[3], aAddress[2], aAddress[1], aAddress[0]);
+
+    RAIL_IEEE802154_SetLongAddress(aAddress);
+}
+
+void otPlatRadioSetShortAddress(otInstance *aInstance, uint16_t aAddress)
+{
+    (void)aInstance;
+
+    otLogInfoPlat(sInstance, "ShortAddr=%X", aAddress);
+
+    RAIL_IEEE802154_SetShortAddress(aAddress);
+}
+
+bool otPlatRadioIsEnabled(otInstance *aInstance)
+{
+    (void)aInstance;
+    return (sState != kStateDisabled);
+}
+
+ThreadError otPlatRadioEnable(otInstance *aInstance)
+{
+    CORE_DECLARE_IRQ_STATE;
+    CORE_ENTER_CRITICAL();
+
+    if (!otPlatRadioIsEnabled(aInstance))
```

Comment:
It generally helps with readability.  Using the error handling macros reduces the number of braces, so that while reading you don't have to keep track of as many contexts, and also makes code more compact and concise.